x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_policy_create_extension/endpoint_policy_create_extension.test.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+import { screen } from '@testing-library/react';
+import { EndpointPolicyCreateExtension } from './endpoint_policy_create_extension';
+import type {
+  NewPackagePolicy,
+  NewPackagePolicyInput,
+  NewPackagePolicyInputStream,
+} from '@kbn/fleet-plugin/common';
+import type { AppContextTestRender } from '../../../../../../common/mock/endpoint';
+import { createAppRootMockRenderer } from '../../../../../../common/mock/endpoint';
+
+const mockNewPackagePolicyInputStream: NewPackagePolicyInputStream = {
+  enabled: true,
+  data_stream: {
+    dataset: 'someDataset',
+    type: 'someType',
+  },
+};
+
+const mockNewPackagePolicyInput: NewPackagePolicyInput = {
+  type: 'someType',
+  enabled: true,
+  streams: [mockNewPackagePolicyInputStream],
+};
+
+const mockNewPackage: NewPackagePolicy = {
+  id: 'someid',
+  inputs: [mockNewPackagePolicyInput],
+  name: 'someName',
+  namespace: 'someNamespace',
+  enabled: true,
+  policy_id: 'somePolicyid',
+};
```

Comment:
I would like to avoid the potential for this data being mutated. Can you convert them to functions that return the mock?